### PR TITLE
Moving IHttpContextFactory to Microsoft.AspNet.Http.Abstractions

### PR DESCRIPTION
**Pull request by JunTaoLuo**
And implementation to Microsoft.AspNet.Http.

### CODE REVIEW

File: src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Http
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs]: .Internal? @muratg this is not for general consumption.

File: test/Microsoft.AspNet.Http.Tests/HttpContextFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            // Assert
+            Assert.True(ReferenceEquals(context, accessor.HttpContext));
+        }
+
+        [Fact]
+        public void Mutable_FeatureCollection_Wrapped_For_OwinFeatureCollection()
+        {
+            var env = new Dictionary<string, object>();
+            var contextFactory = new HttpContextFactory(new HttpContextAccessor());
+            var context = contextFactory.CreateHttpContext(new FeatureCollection(new OwinFeatureCollection(env)));
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Http.Tests/HttpContextFactoryTests.cs]: This test is out of date and no longer relevant, remove it and the dependency.

File: src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs (hunk 1 above)
Comment (muratg) [inline src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs]: Agreed, let's move to .Internal namespace.

File: src/Microsoft.AspNet.Http/HttpContextFactory.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        public HttpContext CreateHttpContext(IFeatureCollection featureCollection)
+        {
+            var httpContext = new DefaultHttpContext(featureCollection);
+            _httpContextAccessor.HttpContext = httpContext;
+            return httpContext;
+        }
+
+        public void Dispose(HttpContext httpContext)
+        {
+
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Http/HttpContextFactory.cs]: Set _httpContextAccessor.HttpContext = null in here

File: test/Microsoft.AspNet.Http.Tests/project.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
   "dependencies": {
+    "Microsoft.AspNet.Owin": "1.0.0-*",
```
Comment (davidfowl) [inline test/Microsoft.AspNet.Http.Tests/project.json]: What @Tratcher said so you can remove this

File: src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNet.Http.Abstractions/IHttpContextFactory.cs]: Noo, leave it public.

**Comment by davidfowl** [comment]
:shipit:

Why is the ci failing

**Pull request merged**